overlord/devicestate/devicectx.go

```diff
@@ -71,3 +78,7 @@ func (dc modelDeviceContext) Store() snapstate.StoreService {
 func (dc modelDeviceContext) ForRemodeling() bool {
```

Comment:
modelDeviceContext has grown enough that we should probably switch to *modelDeviceContext here